api/tests/unit/domain/usecases/authentication/authenticate-oidc-user_test.js

```diff
@@ -0,0 +1,441 @@
+const { expect, sinon, catchErr, domainBuilder } = require('../../../../test-helper');
+const { UnexpectedOidcStateError, UnexpectedUserAccountError } = require('../../../../../lib/domain/errors');
+const logger = require('../../../../../lib/infrastructure/logger');
+
+const authenticateOidcUser = require('../../../../../lib/domain/usecases/authentication/authenticate-oidc-user');
+const AuthenticationSessionContent = require('../../../../../lib/domain/models/AuthenticationSessionContent');
+const AuthenticationMethod = require('../../../../../lib/domain/models/AuthenticationMethod');
+const moment = require('moment');
+
+describe('Unit | UseCase | authenticate-oidc-user', function () {
+  let oidcAuthenticationService;
+  let authenticationSessionService;
+  let authenticationMethodRepository;
+  let userRepository;
+  const externalIdentityId = '094b83ac-2e20-4aa8-b438-0bc91748e4a6';
+
+  beforeEach(function () {
+    oidcAuthenticationService = {
+      identityProvider: 'POLE_EMPLOI',
+      createAccessToken: sinon.stub(),
+      saveIdToken: sinon.stub(),
+      createAuthenticationComplement: sinon.stub(),
+      exchangeCodeForTokens: sinon.stub(),
+      getUserInfo: sinon.stub(),
+    };
+
+    authenticationMethodRepository = {
+      create: sinon.stub(),
+      updateAuthenticationComplementByUserIdAndIdentityProvider: sinon.stub(),
+      findOneByUserIdAndIdentityProvider: sinon.stub(),
+    };
+
+    authenticationSessionService = {
+      save: sinon.stub(),
+    };
+
+    userRepository = {
+      findByExternalIdentifier: sinon.stub(),
+      updateLastLoggedAt: sinon.stub(),
+    };
+  });
+
+  context('When the request state does not match the response state', function () {
+    it('should throw an UnexpectedOidcStateError', async function () {
+      // given
+      const stateSent = 'stateSent';
+      const stateReceived = 'stateReceived';
+      sinon.stub(logger, 'error');
+
+      // when
+      const error = await catchErr(authenticateOidcUser)({
+        code: 'code',
+        redirectUri: 'redirectUri',
+        stateReceived,
+        stateSent,
+        authenticationSessionService,
+        userRepository,
+      });
+
+      // then
+      expect(error).to.be.an.instanceOf(UnexpectedOidcStateError);
+      expect(logger.error).to.have.been.calledWith(
+        `State sent ${stateSent} did not match the state received ${stateReceived}`
+      );
+    });
+  });
+
+  it('should retrieve authentication token', async function () {
+    // given
+    _fakeOidcAPI({ oidcAuthenticationService, externalIdentityId });
+
+    // when
+    await authenticateOidcUser({
+      authenticatedUserId: undefined,
+      clientId: 'clientId',
+      code: 'code',
+      redirectUri: 'redirectUri',
+      stateReceived: 'state',
+      stateSent: 'state',
+      oidcAuthenticationService,
+      authenticationSessionService,
+      authenticationMethodRepository,
+      userRepository,
+    });
```

Comment:
désolé va plus y avoir le 666 👀

```suggestion
    await authenticateOidcUser({
      code: 'code',
      redirectUri: 'redirectUri',
      stateReceived: 'state',
      stateSent: 'state',
      oidcAuthenticationService,
      authenticationSessionService,
      authenticationMethodRepository,
      userRepository,
    });
```